### PR TITLE
EVG-3638: refactor thirdparty S3 to support multipart upload

### PR DESCRIPTION
**Pull request by li-cindy**
Using a `newCode` boolean to keep legacy code around. 
`PutS3File` and `GetS3File` now use the AWS SDK, and `PutS3File` always uploads using multipart upload (https://docs.aws.amazon.com/AmazonS3/latest/dev/mpuoverview.html).

### CODE REVIEW

File: thirdparty/s3.go [hunk 1]
```diff
@@ -179,7 +189,187 @@ func PutS3File(pushAuth *aws.Auth, localFilePath, s3URL, contentType, permission
...
+	// Get size of file to read into buffer
+	fileInfo, err := file.Stat()
+	if err != nil {
+		return errors.Wrapf(err, "Error getting stats for file %s", localFilePath)
+	}
+	size := fileInfo.Size()
+	buffer := make([]byte, size)
+	_, err = file.Read(buffer)
+	if err != nil {
+		return errors.Wrapf(err, "Error reading bytes of file %S", localFilePath)
```
Comment (john-m-liu) [inline thirdparty/s3.go]: lowercase s

File: thirdparty/s3.go [hunk 2]
```diff
@@ -179,7 +189,187 @@ func PutS3File(pushAuth *aws.Auth, localFilePath, s3URL, contentType, permission
...
+	defer file.Close()
+
+	// Get size of file to read into buffer
+	fileInfo, err := file.Stat()
+	if err != nil {
+		return errors.Wrapf(err, "Error getting stats for file %s", localFilePath)
+	}
+	size := fileInfo.Size()
+	buffer := make([]byte, size)
+	_, err = file.Read(buffer)
```
Comment (john-m-liu) [inline thirdparty/s3.go]: Since you're reading the entire file, a simpler way to do this would be to use ioutil.ReadAll

File: thirdparty/s3.go [hunk 3]
```diff
@@ -179,7 +189,187 @@ func PutS3File(pushAuth *aws.Auth, localFilePath, s3URL, contentType, permission
...
+	creationInput := &awsS3.CreateMultipartUploadInput{
+		Bucket:      awsSDK.String(*bucket.Bucket),
+		Key:         awsSDK.String(localFilePath),
+		ContentType: awsSDK.String(contentType),
+	}
+
+	result, err := svc.CreateMultipartUpload(creationInput)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
```
Comment (john-m-liu) [inline thirdparty/s3.go]: What's the purpose of this error handling? It seems like you just return the error

Comment (li-cindy) [reply thirdparty/s3.go]: You're right, I was following their docs with this error handling, but it looks like they were just doing this for ease of printing. I'm going to remove this and just return the error like usual.

File: thirdparty/s3.go [hunk 4]
```diff
@@ -179,7 +188,162 @@ func PutS3File(pushAuth *aws.Auth, localFilePath, s3URL, contentType, permission
...
 
-func GetS3File(auth *aws.Auth, s3URL string) (io.ReadCloser, error) {
+func PutS3File(pushAuth *aws.Auth, localFilePath, s3URL, contentType, permissionACL string) error {
+	if !newCode {
+		return legacyPutS3File(pushAuth, localFilePath, s3URL, contentType, permissionACL)
+	}
+
+	urlParsed, err := url.Parse(s3URL)
+	if err != nil {
+		return err
```
Comment (bsamek) [inline thirdparty/s3.go]: wrap?

**Pull request merged**